frontend/simple/views/Invite.vue

```diff
@@ -164,21 +272,72 @@ export default {
             },
             mailbox
           )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
-
-          // We need to make a record of the invitation in the group's contract
-          const latest = await backend.latestHash(this.$store.state.currentGroupId)
-          const invited = new Events.HashableGroupRecordInvitation(
-            {
-              username: member.state.attributes.name,
-              inviteHash: invite.toHash(),
-              sentDate
-            },
-            latest
-          )
-          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+          // We need to determine if the invitation needs a vote
+          const groupLatest = await backend.latestHash(this.$store.state.currentGroupId)
+          if (this.contract.members.length >= 3) {
+            // Create the Record for the group
+            const invited = new Events.HashableGroupRecordInvitation(
+              {
+                username: member.state.attributes.name,
+                inviteHash: '{lastActionHash}', // these are place holders to be added later
+                sentDate: '{actionDate}'
+              },
+              null
+            )
+            // Create the Proposal and add to it the record and invite actions
+            const proposal = new Events.HashableGroupProposal({
+              proposal: template(L('This is a Vote for {name} to become a member of {group}'),
+                {name: member.state.attributes.displayName || member.state.attributes.name, group: this.contract.groupName}
+              ),
+              // calculate the voting threshold from the group data
+              percentage: this.contract.memberApprovalPercentage,
+              candidate: member.state.attributes.name,
+              actions: [
+               { contractId: member.state.attributes.mailbox, action: JSON.stringify(invite.toObject()) },
+               { contractId: this.$store.state.currentGroupId, action: JSON.stringify(invited.toObject()) }
+              ]
+            }, groupLatest)
+            await backend.publishLogEntry(this.$store.state.currentGroupId, proposal)
+            // send a vote to all members
+            for (let groupMember of this.contract.members) {
```

Comment:
Per convo:

- Split up this invite view into two separate functions (at least, if not two separate components), to handle the separate cases of inviting members to a newly created group vs. making a proposal to invite one new member
- Instead of sending messages to people's mailboxes, it is sufficient to simply publish the invite proposal to the log, along with the fact that the current user is voting for it (so that makes stuff like `if (groupMember === this.$store.state.loggedIn.name)` not necessary too)
- Then on in `handleEvent` and in `simple/js/events.js`, and in the mutation for `HashableGroupRecordInvitation`, add an async action if necessary to do any async things to the UI (might not be necessary if the UI automatically updates by watching the vuex store)